File: app/src/main/java/vergecurrency/vergewallet/viewmodel/SelectCurrencyViewModel.java
package vergecurrency.vergewallet.viewmodel;


import android.app.Application;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import vergecurrency.vergewallet.service.model.Currency;
import vergecurrency.vergewallet.utilities.CurrenciesUtils;


public class SelectCurrencyViewModel extends AndroidViewModel {

	private ArrayList<Currency> currencies;

	public SelectCurrencyViewModel(@NonNull Application application) {
		super(application);

		currencies = CurrenciesUtils.loadCurrenciesFromFile(getApplication().getApplicationContext());
	}

	public ArrayList<Currency> getCurrencies() {
		return currencies;
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/SelectCurrencyViewModel.kt
package vergecurrency.vergewallet.viewmodel


import android.app.Application
import androidx.lifecycle.AndroidViewModel
import vergecurrency.vergewallet.helpers.utils.CurrenciesUtils
import vergecurrency.vergewallet.service.model.Currency
import java.util.*


class SelectCurrencyViewModel(application: Application) : AndroidViewModel(application) {

    val currencies: ArrayList<Currency>?

    init {

        currencies = CurrenciesUtils.loadCurrenciesFromFile(getApplication<Application>().applicationContext)
    }
}
